### PR TITLE
Deëxperiment some recent features in time for 5.40

### PR DESCRIPTION
**Pull request by leonerd**
We recently decided that these four features seem stable enough to remove experimental warnings from.

Three features removed warnings entirely:

 * `sub :const`
 * `foreach my (LIST...)`
 * extra paired delimiters

Additionally, removed experimental warnings from `try/catch` blocks that don't have a `finally` on them. The `finally` itself remains experimental due to complications about exception-during-exception, for the same reason that `defer` does.

### CODE REVIEW

**Comment by leonerd** [comment]
One thing I specifically did not do, is add the now-stable features (`try`, `extra_paired_delimiters`) to the v5.40 feature bundle. It seems a bit too close to release now to consider adding those, in case that might cause extra complications. Hopefully we'll consider adding them to 5.42 though.

File: regen/feature.pl [hunk 1]
```diff
@@ -943,7 +942,16 @@ =head2 The 'extra_paired_delimiters' feature
...
 
-This feature is available starting in Perl 5.36.
+This feature is available starting in Perl 5.36.  From Perl 5.36 to 5.38,
+it was classed as experimental, and Perl emitted a warning for its usage,
+except when explicitly disabled:
+
+    no warnings "experimental::extra_paired_delimiters";
+
+As of Perl 5.40, use of this feature no longer triggers a warning (though the
+C<experimental::extra_paired_delimiters> warning category stilll exists for
```
Comment (Grinnz) [inline regen/feature.pl]:
```suggestion
C<experimental::extra_paired_delimiters> warning category still exists for
```

Comment (leonerd) [reply regen/feature.pl]: Ahyes. Annoyingly, I copied that typo from the `isa` section. I'll fix that up separately.

**Comment by Leont** [comment]
Please do follow up with a experimental.pm PR

**Review by ilmari — APPROVED** [review]

**Pull request merged**